Google.PowerShell.IntegrationTests/Compute/Compute.GceHealthCheck.Tests.ps1

```diff
@@ -0,0 +1,238 @@
+            $healthCheck.UnhealthyThreshold | Should Be 2
+        }
+
+        It "should set values" {
+            $healthCheck = Add-GceHealthCheck $healthCheckName -Description "Test Description" `
+                -HostHeader "google.com" -Port 50 -RequestPath "/some/path" -CheckInterval "0:0:2" `
+                -Timeout "0:0:2" -HealthyThreshold 3 -UnhealthyThreshold 3 -Https
+            ($healthCheck | Get-Member).TypeName | Should Be Google.Apis.Compute.v1.Data.HttpsHealthCheck
+            $healthCheck.Name | Should Be $healthCheckName
+            $healthCheck.Port | Should Be 50
+            $healthCheck.Host | Should Be "google.com"
+            $healthCheck.RequestPath | Should Be "/some/path"
+            $healthCheck.CheckIntervalSec | Should Be 2
+            $healthCheck.TimeoutSec | Should Be 2
+            $healthCheck.HealthyThreshold | Should Be 3
+            $healthCheck.UnhealthyThreshold | Should Be 3
+
+        }
+
+        It "should use an http object over pipeline" {
+            $initHealthCheck = New-Object Google.Apis.Compute.v1.Data.HttpHealthCheck
+            $initHealthCheck.Name = $healthCheckName
+            $healthCheck = $initHealthCheck | Add-GceHealthCheck
+            ($healthCheck | Get-Member).TypeName | Should Be Google.Apis.Compute.v1.Data.HttpHealthCheck
+            $healthCheck.Name | Should Be $healthCheckName
+            $healthCheck.Port | Should Be 80
+            $healthCheck.RequestPath | Should Be "/"
+            $healthCheck.CheckIntervalSec | Should Be 5
+            $healthCheck.TimeoutSec | Should Be 5
+            $healthCheck.HealthyThreshold | Should Be 2
+            $healthCheck.UnhealthyThreshold | Should Be 2
+        }
+
+        It "should use an https object over pipeline" {
+            $initHealthCheck = New-Object Google.Apis.Compute.v1.Data.HttpsHealthCheck
+            $initHealthCheck.Name = $healthCheckName
+            $healthCheck = $initHealthCheck | Add-GceHealthCheck
+            ($healthCheck | Get-Member).TypeName | Should Be Google.Apis.Compute.v1.Data.HttpsHealthCheck
+            $healthCheck.Name | Should Be $healthCheckName
+            $healthCheck.Port | Should Be 443
+            $healthCheck.RequestPath | Should Be "/"
+            $healthCheck.CheckIntervalSec | Should Be 5
+            $healthCheck.TimeoutSec | Should Be 5
+            $healthCheck.HealthyThreshold | Should Be 2
+            $healthCheck.UnhealthyThreshold | Should Be 2
+        }
+    }
+}
+
+Describe "Get-GceHealthCheck" {
+    $healthCheckName = "test-get-gcehealthcheck-$r"
+    $healthCheckName2 = "test-get-gcehealthcheck2-$r"
+
+    It "should fail for wrong project" {
+        { Get-GceHealthCheck -Project "asdf" } | Should Throw 403
+    }
+
+    It "should fail for non-existant health check" {
+        { Get-GceHealthCheck $healthCheckName -Http } | Should Throw 404
+    }
+
+    It "should successfully list zero" {
```

Comment:
"list zero" doesn't make sense without any context. How about "retrieve all health checks for a project".

And the first scenario of the test confirms there are 0, then it adds 2, and confirms the count is then 2.